src/Resources/contao/library/Contao/User.php

```diff
@@ -320,6 +321,8 @@ public function authenticate()
 	 */
 	public function login()
 	{
+		/** @var Request $request */
+		$request = System::getContainer()->get('request_stack')->getCurrentRequest();
```

Comment:
I think it's almost always wrong to take the master request. The use cases are only very, very limited. So no: the current request is the way to go.